src/baal/active/heuristics/heuristics.py

```diff
@@ -295,6 +305,24 @@ def compute_score(self, predictions):
 
         return np.array(history)
 
+    def reorder_indices(self, ranks):
+        """
+        order the indices based on ranks.
+
+        Args:
+            ranks (nd.array): the calculated rank for each sample based
+                on uncertainties.
+
+        Returns:
+            ranks (nd.array): ranks with some noise based on the value for
+                `shuffle_prop`.
+        """
+        if isinstance(ranks, Sequence):
```

Comment:
We could get duplicates there right? Or we assume that the user has done the reindexing.

Reply:
duplicates of what?

Reply:
BatchBald return ranks.
if ranks is a Sequence (so BatchBald was computed by batch), the ids will be according to each batch not the overall heuristic

Reply:
I think we should assume the user has taken care of that, since this function literally gets the final sequence which might be sequence of sequence and it should be taken care of. Also, considering not using chunk processing, anyways BatchBald does batch processing,so how was reindexing taken care of before?